### DISCUSSION TITLE
Trivy Scan Inaccurate Reporting

### DISCUSSION THREAD

**Discussion by upatel2227** [aquasecurity/trivy#4068] [Q&A] [answered]
I have a docker image, built from a dockerfile. The dockerfile removes certain private keys that were found in the original base [aIMAGE] image.  The resulting image [bIMAGE] after the docker build has the keys removed. 

If I scan the image, bIMAGE, using trivy, it still reports the keys still exist (in the same exact location as the orginal base image - aIMAGE.  Am I missing something here?  I also connected into the bash shell of bIMAGE, using docker interactive mode, and did a complete search of the .key files, and it confirmed that they were removed during the dockerfile build job.

**Comment by knqyf263** [comment, marked as answer]
Each layer is independent; files will remain if you remove a file in the upper layer.
